### DISCUSSION TITLE
sBTC Research - Callback on Deposit to Principle

### DISCUSSION THREAD

**Discussion by AshtonStephens** [stacks-archive/sbtc-developer-release#313] [Ideas]
This is a research discussion post meant to be edited with research notes. This research surrounds initiating a clarity contract when a certain principle receives sBTC or stacks.

This may not be relevant to sBTC but it would be good to do some preliminary research.

**Comment by AshtonStephens** [comment]
Currently under investigation by @radicleart

**Comment by friedger** [comment]
Transferring tokens into a contract is (currently) not recommended. Usually, depositing happens through a defined trait.
However, sbtc as of now does not provide anything to specify a trait.

Specifying a trait would cost more on the bitcoin side when making the request.

Could there be a default trait? What happens if the contract call fails? Would it be one mint tx and one callback tx? Who covers the tx fees for the second call?

Alternatively, is it possible to lookup the deposit/mint details in the sbtc contract?

**Comment by mijoco-btc** [comment]
there will always be an event generated by successful mint so maybe its
safer to let the contract team listen for events and call contract

more decoupled and no risk of the mint failing because of the third party
contract.

**Comment by AshtonStephens** [comment]
In conclusion, we've decided that this feature should be implemented eventually, but isn't going to be part of the sBTC release.
